src/gentropy/dataset/study_index.py

```diff
@@ -590,3 +592,158 @@ def annotate_sumstats_qc(
+                        f.collect_list("analysisFlags").over(full_study_id_window)
+                    ),
+                )
+                # Propagating hasSumstatsFlag - if no flag, leave null:
+                .withColumn(
+                    "hasSumstats",
+                    f.when(
+                        # There's a true:
+                        f.array_contains(
+                            f.collect_set("hasSumstats").over(full_study_id_window),
+                            True,
+                        ),
+                        f.lit(True),
+                    ).when(
+                        # There's a false:
+                        f.array_contains(
+                            f.collect_set("hasSumstats").over(full_study_id_window),
+                            False,
+                        ),
+                        f.lit(False),
+                    ),
+                )
+                # Propagating disease: when different sets of diseases available for the same study,
+                # we pick the shortest list, becasuse we assume, that is the most accurate disease assignment:
+                .withColumn(
+                    "mostGranular",
+                    f.size(f.col("traitFromSourceMappedIds"))
+                    == f.min(f.size(f.col("traitFromSourceMappedIds"))).over(
+                        full_study_id_window
+                    ),
+                )
+                # Remove less granular disease mappings:
+                .withColumn(
+                    "traitFromSourceMappedIds",
+                    f.when(f.col("mostGranular"), f.col("traitFromSourceMappedIds")),
+                )
+                # Propagate mapped disease:
+                .withColumn(
+                    "traitFromSourceMappedIds",
+                    f.last(f.col("traitFromSourceMappedIds"), True).over(
+                        full_study_id_window
+                    ),
+                )
+                # Repeating these steps for the `traitFromSource` column:
+                .withColumn(
+                    "traitFromSource",
+                    f.when(f.col("mostGranular"), f.col("traitFromSource")),
+                )
+                # Propagate disease:
+                .withColumn(
+                    "traitFromSource",
+                    f.last(f.col("traitFromSource"), True).over(full_study_id_window),
+                )
+                # Distinct study types are joined together into a string. So, if there's ambiguite, the study will be flagged when the study type is validated:
+                .withColumn(
+                    "studyType",
+                    f.concat_ws(
+                        ",", f.collect_set("studyType").over(full_study_id_window)
+                    ),
+                )
+                # At this point, all studies in one window is expected to be identical. Let's just pick one:
+                .withColumn("rank", f.row_number().over(study_id_window))
```

Comment:
This is a quiet complex set of window operations. I wonder if it would not be easier to follow:
- anti join pics and susie and rejoin difference to susie
- anti join top_hits and sumstats and rejoin difference to sumstats